lib/widgets/scaffold.dart

```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
       ),
     );
   }
+
+  ///
+  /// Build content body based on is web version.
+  /// When web version ad an action bar to right side.
+  ///
+  Widget buildBody(BuildContext context, {bool enableContent = false}) {
+    final List<Widget> actionButtons =
+        !kIsWeb // check to avoid building action button twice in mobile
+            ? List.empty()
+            : ([leading] + actions)
+                .map((action) => buildMaterialActionButton(action, context,
+                    enableContent: enableContent))
+                .where((element) => element != null)
+                .toList();
+
+    return kIsWeb && actionButtons.isNotEmpty
+        ? Stack(
+            children: [
+              Center(
+                  child: ConstrainedBox(
+                      constraints: BoxConstraints(maxWidth: maxBodyWidth),
+                      child: body)),
+              ActionSideBar(actionButtons),
+            ],
+          )
+        : body;
+  }
+
+  ///
+  /// Wrap simple action button to enable hover and splash effects
+  ///
+  AspectRatio buildMaterialActionButton(
```

Comment:
This feels like it should be the sidebar's job. Why not make the sidebar constructor take a list of AppScaffoldAction as a parameter instead?

Reply:
Losing access to private function _widgetFromAction but could pass widgets from actions and inside sidebar class to wrap into material.

Reply:
Moved to action sidebar class:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R187-R191

Reply:
What if we move _widgetFromAction to AppScaffoldAction and make it a public method, like `toWidget({bool enabled, BuildContext context})`?

So you could just call the sidebar like:
```dart
ActionSideBar(actions: [leading] + actions, enabled: enableContent);
```

Reply:
Yes, definitely don't belongs to  AppScaffold.

Reply:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/49d688c89623ace6d8c3826d72aaaec2f17da1aa